FundParser/DAL/Repository/Repository.cs

```diff
@@ -29,7 +32,8 @@ public virtual async Task<TEntity> Insert(TEntity entity, CancellationToken canc
 
         public virtual async Task Delete(int id, CancellationToken cancellationToken = default)
         {
-            TEntity entityToDelete = await _dbSet.FindAsync([id], cancellationToken: cancellationToken) ?? throw new Exception("Entity with given Id does not exist.");
+            TEntity entityToDelete = await _dbSet.FindAsync([id], cancellationToken: cancellationToken)
```

Comment:
It should be possible to call `GetByID` here to not duplicate the exception and its message